running-your-first-ot-server/setup-webserver-linux.md

```diff
@@ -0,0 +1,201 @@
+```bash
+server {
+        listen 80 default_server;
+        listen [::]:80 default_server;
+
+        root /var/www/html;
+
+        # Add index.php to the list if you are using PHP
+        index index.php index.html index.htm index.nginx-debian.html;
+
+        server_name _;
+
+        location / {
+                # First attempt to serve request as file, then
+                # as directory, then fall back to displaying a 404.
+                try_files $uri $uri/ =404;
+        }
+
+        # pass PHP scripts to FastCGI server
+        #
+        location ~ \.php$ {
+               include snippets/fastcgi-php.conf;
+        #
+        #       # With php-fpm (or other unix sockets):
+        ################## UBUNTU 18.04 LTS ##################
+               fastcgi_pass unix:/var/run/php/php7.2-fpm.sock;
+        ################## UBUNTU 20.04 LTS ##################
+        #       fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
+
+        }
+
+}
+
+```
+Now we have to save the file and quit the file, we do this the following way:
+`CTRL+C` After that you will just write `:wq` press enter and we have saved and quit the file :)
+
+We are set with the entire webserver now and so its time to check if we actually did everything as we supposed to do, but first we have to restart our web features. We do this with the following commands:
+`Change the number of the php-fpm version to one you're using!!!`
+```bash
+sudo systemctl restart nginx
+sudo systemctl restart php7.2-fpm
+```
+Lets open our browser and goto your ip-address and see how wonderful you are!
+Your very own setup and configured server!
+
+# Setup the database user + phpmyadmin
+
+We need to create ourselves a mysql admin account. We will be doing this the following way. Please follow the steps carefully :)
+
+```bash
+sudo mariadb
+```
+![alt text](https://worldofcoding.net/github-img/otland-gitbook/phpversion.png "phpversion")
+###### You are inside the database right now!
+
+We will be creating the user 'otadmin' with the password of 'otadminpassword', change these values to what you want it to be!
+```bash
+create user 'otadmin'@'localhost' identified by 'otadminpassword';
+grant all privileges on *.* to 'otadmin'@'localhost';
+flush privileges;
+exit;
+```
+Insert these lines one by one.
+Now we have setup our database superuser :)
+
+## Configure phpmyadmin
+Setup phpmyadmin to access your database via the browser.
+
+Insert the following configuration in a new file inside the nginx folder. The following command will create the file and when you are in insert the configuration posted below.
+( same as for the webserver nginx configuration I have made comments for the ubuntu versions inside the php location )
+`sudo vim /etc/nginx/sites-available/phpmyadmin`
+```bash
+server {
+    listen      2344;
```

Comment:
Why this port?

Reply:
Just took a random port, can be changed to what you desire.